temporallib/auth/auth.py

```diff
@@ -22,43 +23,68 @@ class MacaroonAuthOptions:
     Defines the parameters for authenticating with Candid.
     """
 
-    macaroon_url: str
-    username: str
-    keys: KeyPair
+    macaroon_url: str = None
+    username: str = None
+    keys: KeyPair = None
+
+    def __post_init__(self):
+        self.macaroon_url = self.macaroon_url or os.getenv("TEMPORAL_CANDID_URL")
+        self.username = self.username or os.getenv("TEMPORAL_CANDID_USERNAME")
+        self.keys = self.keys or KeyPair()
+
 
 
 @dataclass
 class GoogleAuthOptions:
     """
     Defines the parameters for authenticating with Google IAM.
     """
+    type: str = None
+    project_id: str = None
+    private_key_id: str = None
+    private_key: str = None
+    client_email: str = None
+    client_id: str = None
+    auth_uri: str = None
+    token_uri: str = None
+    auth_provider_x509_cert_url: str = None
+    client_x509_cert_url: str = None
+
+    def __post_init__(self):
+        self.type = self.type or os.getenv("TEMPORAL_OIDC_AUTH_TYPE")
+        self.project_id = self.project_id or os.getenv("TEMPORAL_OIDC_PROJECT_ID")
```

Comment:
[suggestion] You can use Pydantic here.  I built a similar structure for [airbyte-tranformation workflow](https://github.com/canonical/cs-workflows/blob/main/workflows/airbyte_transformation/airbyte_transformation/common/app_settings.py).

Reply:
With that, you can also use `Optional` keyword.